system/lib/libc/emscripten_memset.c

```diff
@@ -5,4 +5,20 @@
 #endif
 #endif
 
+#ifdef EMSCRIPTEN_OPTIMIZE_FOR_OZ
+
+#include <stddef.h>
+
+void *memset(void *str, int c, size_t n)
```

Comment:
nit: We are trying to move towards a single coding style based on the LLVM style (see .clang-format at the top level).. which means braces on the same line as the decl.

Reply:
Not a fan of changing code style on the grounds of LLVM, but if it helps, I don't mind.

Reply:
Are you not a fan of the LLVM style or not a fan of having a style in general?

IIRC I chose LLVM because it what we use in upstream LLVM (obviously) and also in binaryen and similar to the chromium style used in wabt.   So it just what most of the team is use to.

I'm up for discussing what style we should use. but I don't like he idea of not having any consistent style at all.  I also like the idea of using clang-format to avoid extended style discussions and making it super easy to conform to the style without adding work... so I'm hoping we can do that here too.

Reply:
Not a fan of the { on decl line, I don't think we followed that for .c code before.

The general issue I have with styling is that it has lead to all sorts of checkers - clang-format and flake and js-lint tools added to the repo. They are making contributing from Windows (and on macOS, because of the MacPorts vs Homebrew vs whatever you have installed manually vs macOS version - conflict hell) more of a barrier to entry than it used to be. The tooling may be practically seamless for Linux, but on Windows and on macOS one needs to hunt down installation of tools that are not there by default.

Clang-format on Visual Studio is particularly annoying: it hangs VS often, it formats right in the middle of you typing, it repeatedly brings in popups "hey we notice that you have a clang-format in this repo so we just reformatted for you", so that using VS to edit any files in Emscripten repo has become a dont-do-that thing.

That is why I see that these kinds of styling requirements generally add more work than is worth. I do wish I could also use Linux all day and not care about Windows and macOS, but that is where our users are, hence needing to make sure things roll fine on those OSes.

Reply:
I see you point.  If some developers find it hard to run these tools then I agree we should not make it hard requirement in this repo.

However, I don't think that means we should not have a preferred style for the code in the repro or that we should not make a best effort to conform to it where possible/easy.

Reply:
Per the above I'm not blocking this from landing based on style here.  I leave it up to you.